### PR TITLE
get spot by ranges of x and y

### PR DESCRIPTION
**Pull request by tsukasa2**
I added "xMax", "xMin", "yMax" and "yMin" to keyword to search spots.
To test it on Postman,
- Request URL: GET http://localhost:5000/spot/getSpot
- Body: JSON, for example,
```
{
    "spotType": "shopping",
    "xMax": 180,
    "xMin": -180,
    "yMax": 34,
    "yMin": 0
}
```

### CODE REVIEW

**Review by ShimizuKazuki — CHANGES_REQUESTED** [review]
{
    "spotType": "shopping",
    "xMin": 50,
    "xMax": 40
}
=>
{
    "success": false,
    "error": "spot does not exist"
}

x, yの範囲に条件つけたほうがいいかも

**Comment by tsukasa2** [comment]
常に-180 <= xMin <= xMax <= 180かつ-90 <= yMin <= yMax <= 90となるように修正しました．

File: back/src/database/makeSQL.js [hunk 1]
```diff
@@ -16,6 +16,53 @@ module.exports.getSpotQueryBuilder = function(keywords){
...
+    }
+    if( keywords.yMin != null ){
+        var yMin = Math.max( keywords.yMin, -90 );
+        yMin = Math.min( yMin, 90 );
+    }
+
+    // confirm that xMax >= xMin and that yMax >= yMin
+    if( keywords.xMax != null && keywords.xMin != null ){
+        if( xMax < xMin ){
+            console.log("xMin is bigger than xMax.");
```
Comment (ShimizuKazuki) [inline back/src/database/makeSQL.js]: ログはただの標準出力より
error(fileLabel,"Error while saving spot: " + err);
こんな感じにしたほうがいい

File: back/src/database/makeSQL.js [hunk 2]
```diff
@@ -16,6 +16,53 @@ module.exports.getSpotQueryBuilder = function(keywords){
...
+            xMax = 180;
+            xMin = -180;
+        }
+    }
+    if( keywords.yMax != null && keywords.yMin != null ){
+        if( yMax < yMin ){
+            console.log("yMin is bigger than yMax.");
```
Comment (ShimizuKazuki) [inline back/src/database/makeSQL.js]: here too

**Review by ShimizuKazuki — CHANGES_REQUESTED** [review]

**Comment by ONourry** [comment]
This request should use the new empty function from #62

**Comment by tsukasa2** [comment]
isEmpty( X ) returns true if X is 0. However, xMax, xMin, yMax and yMin can be 0 in this case, I think.

**Comment by ONourry** [comment]
It would be more efficient to modify isEmpty() and allow value 0 than adding extra || value == 0 everywhere.

**Comment by tsukasa2** [comment]
I modified isEmpty( X ) to return FALSE when X is 0.
( The commit massage is wrong ... 🤦  )

**Review by ShimizuKazuki — APPROVED** [review]

**Pull request merged**